### DISCUSSION TITLE
documentation for v5 and should keyword

### DISCUSSION THREAD

**Discussion by tiec7** [chaijs/chai#1571] [Support]
I'm new with chai, and I was following a tutorial; now of course all the tutorials does not work fine with v5, but I managed to let my super-base test work, using expect:

`package.json`
```
{
  "name": "firsttest",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "chai": "^5.0.0",
    "mocha": "^10.2.0",
    "selenium-webdriver": "^4.16.0"
  }
}
```

`testfile.js`
```
import { Builder, By, Key } from 'selenium-webdriver'
import { assert, expect, should } from 'chai'

async function example() {
  // launch the browser
  let driver = await new Builder().forBrowser('chrome').build()

  // navigate to app
  await driver.get('https://lambdatest.github.io/sample-todo-app/')

  // add a todo
  await driver.findElement(By.id('sampletodotext')).sendKeys('learn selenium', Key.RETURN)

  // assert
  let todoText = await driver.findElement(By.xpath('//li[last()]')).getText()    // contains 'learn selenium'

  assert.strictEqual(todoText, 'learn selenium')    // works

  todoText.should.equal('learn selenium')    // DOES NOT WORK

  // close the browser
  await driver.quit()
}

example()
```

Now the `should` keyword in import is grayed out (value declared but never read), and the expression with should throws an error
```
TypeError: Cannot read properties of undefined (reading 'equal')
```

So, how can I manage to work this?

And more generally, is there a place with the new documentation for v5? nothing seems to work from the old one for a beginner...

Maybe because I'm just a beginner, ok, but...

**Comment by official-burak** [comment]
`should` is a function that installs a getter on `Object.prototype`. Importing it does not do that, which is why the named import is unused and `todoText.should` is undefined.

Call it once:

```js
import { should } from 'chai'
should()

todoText.should.equal('learn selenium')
```

Or load the register helper so you do not call it yourself:

```js
import 'chai/register-should'
```

v5 docs: https://www.chaijs.com/guide/styles/ (`Should`). `assert` and `expect` work without this step. `should` does not.
